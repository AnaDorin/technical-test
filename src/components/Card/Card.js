import React from 'react';

const Card = ({index, props}) => {
    const { name, productImageUrl, productImageAltText } = props;
    const price = props.price.formattedValue;

     return (
         <div id={index}>
             <img src={productImageUrl} alt={productImageAltText}/>
             <div>
                 <p>
                     {name}
                 </p>
                 <p>
                     {price}
                 </p>
             </div>
         </div>
     )
};

export default Card;