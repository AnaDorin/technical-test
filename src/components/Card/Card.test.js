import React from "react";
import Card from "./Card";
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

describe("Question component", () => {
    const sliderObject = {
        name: "Digital isolator SO-16W",
        productImageUrl: "https://www.distrelec.biz/Web/WebShopImages/landscape_medium/ho/to/SO-16W-Photo.jpg",
        productImageAltText: "Image not found",
        price: {
            currency: "GBP",
            formattedValue: "8.83"
        },
    }
    const wrapper = () => {
        return render(<Card index={0} props={sliderObject}/>);
    };
    it("should build the component", () => {
        const { container } = wrapper();
        expect(container).toMatchSnapshot();
    });
});