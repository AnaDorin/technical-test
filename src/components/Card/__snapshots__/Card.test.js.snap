// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question component should build the component 1`] = `
<div>
  <div
    id="0"
  >
    <img
      alt="Image not found"
      src="https://www.distrelec.biz/Web/WebShopImages/landscape_medium/ho/to/SO-16W-Photo.jpg"
    />
    <div>
      <p>
        Digital isolator SO-16W
      </p>
      <p>
        8.83
      </p>
    </div>
  </div>
</div>
`;
